### ISSUE TITLE
[BUG] Unable to save SMTP settings on New Fresh Mirth 4.0.1 install on Windows 2022

### ISSUE THREAD

**Issue by Rippenator** [nextgenhealthcare/connect#5228]
[Mirth Log.txt](https://github.com/nextgenhealthcare/connect/files/8827511/Mirth.Log.txt)

I have setup two new Servers on Windows 2022 using Mirth Connect 4.0.1
The problem exists on both servers.  I am not able to save the SMTP information.  After I enter in data I can send a test message, but after I refresh information is blank and log has errors 
 
 
**To Reproduce**
Under Settings 
Add SMTP Host
Add SMTP Port
Set Default From Address. 
Send test message.  (Works)
Refresh and settings are gone
Look at server Log
 
See above
 
**Expected behavior**
The SMTP values save in the databse
 
**Actual behavior**
The SMTP values are not saved.
 
**Screenshots**
Attached Log
 
**Environment (please complete the following information):**
 - OS: Windows 2022
 - Java Distribution/Version jdk-18_windows-x64
 - Connect Version 4.0.1
 
**Workaround(s)**
No Workarounds
 
**Additional context**

**Comment by pacmano1** [comment]
Do both these servers use the same mirthDB?

**Comment by Rippenator** [comment]
Yes they do.

~Jeff R. Ripp

**Comment by Rippenator** [comment]
Yes they do, but  with different schemas.


~Jeff R. Ripp

**Comment by pacmano1** [comment]
I suspect for Nextgen to look at this you will need to provide more specifics.  I have used the same db server with different named databases for each mirth instance.   I have not used the same database with unique schemas for miirth's data store (so sharing your db connect string might help).  The error suggests the row already exissts in the table and it came from somwhere, either the other engine or the engine your are on and there is some problem in mirth or permissions.   If you confirm that row is empty before your start and not empty after you save it and so on, that would help also.

**Comment by Rippenator** [comment]
This was from our DBA.  
I can confirm that the row was empty before you started the install and not empty after you saved it. You didn’t run any other Mirth installs in test, so I don’t think the row came from “the other engine”. So it had to come from the engine you are on. If the problem was permissions you wouldn’t have been able to insert anything into the table, or even create the table. So I think the problem is in Mirth.
The scary thing is that they say they have never used the same database with different schemas for Mirth’s data stores. Is that no longer possible with the upgraded version of Mirth? It works fine in both test and prod in the old version.

**Comment by pacmano1** [comment]
What DB engine?  And I assume you have added the extra vmoptions required for java 9+.   Share the mirth.properties jdbc connection string for both servers (remove private stuff of course).

**Comment by Rippenator** [comment]
The DB engine is Oracle (version 19.14.0.0.0 Enterprise edition).
It’s the same for both the test and dev Mirth engines. Only the database schema name is different.

jdbc:oracle:thin:@slhtora-a:1521/TEST21

Also I did add the extra vmoptions.  Without them it would not connect.

**Comment by pacmano1** [comment]
`The JDBC connection string is the thing Mack sent you.`   huh?

Anyway, it's Oracle, I'm out.

**Comment by Rippenator** [comment]
@pacmano1  Sorry I forgot to edit the message before I posted it.

**Comment by pacmano1** [comment]
No worries at all.  I just don't touch Oracle.

**Comment by Rippenator** [comment]
### This is the error that is causing the problem

ERROR 2022-06-02 12:53:17,801 [ConfigurationServlet Thread (Set server settings) < qtp1381671334-36] com.mirth.connect.server.controllers.DefaultConfigurationController: Could not store property: category=core, name=smtp.password
org.apache.ibatis.exceptions.PersistenceException: 
Error updating database.  Cause: java.sql.SQLIntegrityConstraintViolationException: ORA-00001: unique constraint (MIRTHTEST3.SYS_C0023196) violated

**Comment by ab-20-m** [comment]
The issue is a combination of Oracle treating empty-string as null and how the DefaultConfigurationController.saveProperty(category,name,value) assumes a null results from DefaultConfigurationController.getProperty(category, name) [means the record doesn't exist. ](https://github.com/nextgenhealthcare/connect/blob/74e70c466d1ec72e19ae020e893210a40f037d9b/server/src/com/mirth/connect/server/controllers/DefaultConfigurationController.java#L990-L994)

The workaround would be to manually set some value in the database records for smtp.passsword and smtp.user properties and the values should then stick. The same mis-behavior can be observed for the following settings:

- core environment.name
- core.[UUID] server.name
- core loginnotification.mesage

**Comment by Rippenator** [comment]
Yeah that is what we had to do for a workaround, but it then involves getting the DBA to make changes.  Is this by design?


~Jeff R. Ripp
